dipy/denoise/noise_estimate_localpca.py

```diff
@@ -0,0 +1,111 @@
+"""
+===========================================
+Estimate Noise Levels for LocalPCA Datasets
+===========================================
+
+"""
+
+import numpy as np
+import nibabel as nib
+import scipy as sp
+from scipy import ndimage
+
+def estimate_sigma_localpca(data,gtab):
+    
+    # first identify the number of the b0 images
+    K = gtab.b0s_mask[gtab.b0s_mask].size
+
+    if(K>1):
+        print("MUBE Estimate")
+        # If multiple b0 values then 
+        data0 = data[...,gtab.b0s_mask]
+        # MUBE Noise Estimate
+
+        # Form their matrix
+        X = data0.reshape(data0.shape[0] * data0.shape[1] * data0.shape[2], data0.shape[3])
+        # X = NxK
+        # Now to subtract the mean
+        M = np.mean(X,axis = 0)
+        X = X - M
+        C = np.transpose(X).dot(X)
+        # C = C/data0.shape[3]
+        # Do PCA and find the lowest principal component
+        [d,W] = np.linalg.eigh(C)
+        d[d < 0] = 0;
+        d_new = d[d != 0]
+        # we want eigen vectors of XX^T so we do
+        V = X.dot(W)
+        # unit normalize the clolumns of V
+        # V = V / np.linalg.norm(V, ord=2, axis=0)
+        # As the W is sorted in increasing eignevalue order, so is V
+        # choose the smallest positive principle component
+        I = V[:,d.shape[0] - d_new.shape[0]].reshape(data.shape[0], data.shape[1], data.shape[2])
+        
+
+    else:
+        print("SIBE Estimate")
+        # if only one b0 value then
+        # SIBE Noise Estimate
+        data0 = data[...,~gtab.b0s_mask]
+        # MUBE Noise Estimate
+
+        # Form their matrix
+        X = data0.reshape(data0.shape[0] * data0.shape[1] * data0.shape[2], data0.shape[3])
+        # X = NxK
+        # Now to subtract the mean
+        M = np.mean(X,axis = 0)
+        X = X - M
+        C = np.transpose(X).dot(X)
+        # C = C/data0.shape[3]
+        # Do PCA and find the lowest principal component
+        [d,W] = np.linalg.eigh(C)
+        d[d < 0] = 0;
+        d_new = d[d != 0]
+        # we want eigen vectors of XX^T so we do
+        V = X.dot(W)
+        # unit normalize the clolumns of V
+        # V = V / np.linalg.norm(V, ord=2, axis=0)
+        # As the W is sorted in increasing eignevalue order, so is V
+        # choose the smallest principle component
+        I = V[:,d.shape[0] - d_new.shape[0]].reshape(data.shape[0], data.shape[1], data.shape[2])
```

Comment:
These last lines of code are the same for if(K >1), so I will suggest you to take these out of the if/else statement. The only think that have to be in the statement is 

if(K > 1):
    data0 = data[..., gtab.b0s_mask]
else:
    data0 = data[..., ~gtab.b0s_mask]

right?